# Introduction
A pipeline was created that undertook variant-calling on an ngs data set obtained from tumor samples.
Tumor normal analysis was performed to identify cancer-specific mutations (this was by a custom script, see below).
The tumor-mutational burden was calculated. 

## Targeted variant-calling pipeline
A very simple overview of the Variant Calling pipeline

#### Input Files:  
- `fq` files from tumor samples
- human genome reference `fa` file
- A file of hereditary variants from `gnomAD`
- `bed` files specifying target regions
- annotation files like from `COSMIC`, `1000G`
- configuration file described below

#### Steps
Note that the pipeline was run on a cluster, qsub jobs were used.
- Run QC on data e.g. by `fastqc`. This provided basic statistics like sequence quality, GC content, sequence duplication 
levels, kmer content, and more.
- Run `BWA` alignment
- Do primer soft clipping using amplicon files
- Run `SAMTools` to remove poorly mapped/unmapped reads
- Run `GATK` scripts for indel realignment, base recalibration, variant calling, haplotype calling etc.
- Run `GATK` variant filtration, and add annotations like from COSMIC vcf files
- Run custom script to compare `vcf` file generated by the pipeline against file of known 
hereditary variants, e.g. `gnomAD` data. Remove the hereditary variants
- Run `snpEff` to add annotations e.g. from `COSMIC`
- Perform tmb calculation
- Results are made available through a web UI.

Note there are several other steps along the way as:
- Filtration steps for base/read qualities, coverage, strand bias etc. 
- Additional QC steps recording various statistics


### Configuration
The pipeline was highly configurable. There was a file which allowed users to pick various parameters.
Just a few of these

Options for filtering on values e.g.
  - base quality
  - coverage
  - snp_frequency 

General parameters e.g.
  - genome version
  - annotation files e.g.1000G, cosmic, omniK etc.
  - whether single or paired end reads
  - what should be shown in annotated file columns
  - num threads used when running the BWA alignment

GATK options
  - mismatch values when running realignment
  - genotype likelihood models used by UnifiedGenotyper
  - strand bias